forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java

```diff
@@ -1159,6 +1162,11 @@ else if (origin.contains(ZoneType.Hand) && player.isOpponentOf(decider)) {
                             fetchList = CardLists.getValidCards(fetchList, "Card.powerLE" + totpower, source.getController(), source, sa);
                         }
                     }
+                    if (totalCardTypes != null) {
+                        if (totCardTypes >= 0) {
+                            fetchList = CardLists.getValidCards(fetchList, "Card.numTypesLE" + totCardTypes, source.getController(), source, sa);
```

Comment:
this doesn't seem right
imo you need at least "four or more card types among them" to exile anything
but your logic mirrors the other params which are meant for "or less"

Reply:
Yea, maybe this filter just doesn't matter. Lemme fiddle with it some more.

Reply:
I guess it depends on how the GUI support should behave
in theory just ChooseCard any and then a Condition for ChangeZone would already work 🤔

Reply:
Right, ideal case is counting the types like we do for Mount. But functionality, We don't need as much. Maybe i don't even need engine changes.